src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -39,17 +40,26 @@
 /* controls use of locks to enforce safe commutativity */
 bool AllModificationsCommutative = false;
 
+static uint64 shardForTransaction = INVALID_SHARD_ID;
+static bool callbackInstalled = false;
```

Comment:
I'll look into whether this state variable could be removed somehow. If not, I'll add `xact` to the name.